[AY1920S1-CS2113T-W17-2] WalletCLi

GeraldLimWeeKoon opened the pull request:
@Xdecosee
@kyang96
@matthewng1996
@A0171206R

Commit:
Setting up Import/Export Env

seanieyap commented on src/main/java/wallet/command/Command.java:
```diff
@@ -0,0 +1,156 @@
+                    taskList.modifyTask(num, task);
+                    System.out.println("Nice! I've marked this task as done:");
+                    System.out.println(task.toString());
+                    //fileIO.updateToFile(task, num);
```
Don't leave commented out code. Just go ahead and delete unused code. Now that we are using a revision control system, we can recover any deleted code later.

seanieyap commented on src/main/java/wallet/command/Command.java:
```diff
@@ -0,0 +1,156 @@
+                    taskList.addTask(task);
+                    System.out.println("Got it. I've added this task:");
+                    System.out.println(task.toString());
+                    System.out.println("Now you have " + taskList.getTaskListSize() + " tasks in the list.");
```
Avoid the use of magic string/numbers. Save these messages in a central location to be used by multiple classes/methods and etc. Would save trouble when writing tests as well.

seanieyap commented on src/main/java/wallet/logic/command/DoneCommand.java:
```diff
@@ -0,0 +1,61 @@
+    public static final String MESSAGE_FAILURE_NO_SUCH_ID = "There is no such id in loans!";
+    public static final String MESSAGE_FAILURE_ALREADY_DONE = "The loan has already been settled!";
+
+    private int id;
```
Could be private final?

seanieyap commented on src/main/java/wallet/logic/command/ListCommand.java:
```diff
@@ -0,0 +1,91 @@
+        default:
+            System.out.println(MESSAGE_USAGE);
+        }
+        return false;
```
Indentation issue with switch statement.

seanieyap replied:
Perhaps the use of if statement would be easier to read and neater here. Just as the other classes make use of if statement instead of switch.

seanieyap commented on src/main/java/wallet/logic/command/ListCommand.java:
```diff
@@ -0,0 +1,91 @@
+            if (!isListAll) {
+                break;
+            }
+            //else fallthrough
```
Unnessary comment. Use comments sparingly.

seanieyap commented on src/main/java/wallet/logic/command/SetBudgetCommand.java:
```diff
@@ -0,0 +1,97 @@
+                    return false;
+                } else if (budgetList.getBudgetList().size() != 0 && budget.getAmount() > 0) {
+                    int index = checkDuplicates(wallet.getBudgetList());
+                    if (index != -1) { //There is a duplicate
```
Unnesssary comment.

seanieyap commented on src/main/java/wallet/logic/parser/AddCommandParser.java:
```diff
@@ -0,0 +1,208 @@
+            isLend = false;
+        }
+
+        ArrayList<Contact> contactList = LogicManager.getWallet().getContactList().getContactList();
```
.getContactList().getContactList()? Not just a naming issue but perhaps this can be done more efficiently with reduce dependencies.

seanieyap commented on src/main/java/wallet/logic/parser/ContactParserHelper.java:
```diff
@@ -0,0 +1,172 @@
+     * @return Index of command line option.
+     */
+    public int findIndexOf(String[] info, String toFind) {
+
```
Use blank lines sparingly.

seanieyap commented on src/main/java/wallet/logic/parser/ContactParserHelper.java:
```diff
@@ -0,0 +1,172 @@
+        detailsIndex = findIndexOf(info, "/d");
+        phoneIndex = findIndexOf(info, "/p");
+
+        //Process input into three array list
```
Unnessary comment.

seanieyap commented on src/main/java/wallet/logic/parser/ParserManager.java:
```diff
@@ -0,0 +1,66 @@
+        case AddCommand.COMMAND_WORD:
+            AddCommandParser addCommandParser = new AddCommandParser();
+            return addCommandParser.parse(arguments[1]);
+
```
Unnessary empty lines.

seanieyap commented on src/main/java/wallet/model/contact/ContactList.java:
```diff
@@ -0,0 +1,146 @@
+        return null;
+    }
+
+    /**
+     * Get the current number of contacts in the contactList.
+     *
+     * @return The number of contacts in the list.
+     */
```
Trivial methods do not need header comments. There is an issue of long classes where there are more comments than actual code in this project.

seanieyap commented on src/main/java/wallet/model/record/RecordList.java:
```diff
@@ -0,0 +1,105 @@
+        } catch (ArrayIndexOutOfBoundsException e) {
+            System.out.println("☹ OOPS!!! The date/time of a " + type + " cannot be empty");
+        }
+    }*/
```
Don't leave commented out code. Just go ahead and delete unused code. Now that we are using a revision control system, we can recover any deleted code later.

seanieyap commented on src/main/java/wallet/storage/HelpStorage.java:
```diff
@@ -0,0 +1,98 @@
+
+
+    }
+}
```
Huge issue of blank lines in this class.

seanieyap commented on src/main/java/wallet/thread/MyThread.java:
```diff
@@ -0,0 +1,74 @@
+import wallet.model.record.LoanList;
+import wallet.ui.Ui;
+
+public class MyThread implements Runnable {
```
MyThread is not an appropriate name.

seanieyap commented on src/main/java/wallet/thread/MyThread.java:
```diff
@@ -0,0 +1,74 @@
+    private Ui ui;
+
+    /**
+     * Constructs a custom thread.
```
what is "custom" thread?

seanieyap commented on src/main/java/wallet/thread/MyThread.java:
```diff
@@ -0,0 +1,74 @@
+     * @param loanList      The LoanList object.
+     * @param timeInSeconds The time in seconds.
+     */
+    public MyThread(boolean autoRemind, LoanList loanList, int timeInSeconds) {
```
Missing @param

seanieyap commented on src/main/java/wallet/thread/MyThread.java:
```diff
@@ -0,0 +1,74 @@
+        this.loanList = loanList;
+        this.ui = new Ui();
+        thread = new Thread(this);
+        thread.start(); // Starting the thread
```
Unnecessary comment.

seanieyap commented on src/main/java/wallet/thread/MyThread.java:
```diff
@@ -0,0 +1,74 @@
+
+    /**
+     * Executes the thread.
+     */
```
Unnecessary comment that does not provide any useful information. Perhaps state what the thread does instead.

seanieyap commented on src/main/java/wallet/ui/Notifications.java:
```diff
@@ -0,0 +1,5 @@
+package wallet.ui;
+
+public class Notifications {
+
```
??

seanieyap reviewed (commented):
Overall good work. Most of the software engineering principles are adhered to.

There is, however, an issue of over commenting in this project whereby trivial methods are also described. Also, formatting issues like many blank lines and indentation. The parsing is also alittle of concern; there are some duplicate code that can be reduced by applying SLAP. Please use the final modifier when possible and reduce the use of magic string and numbers.

Please close this PR after my comments have been read.

Commit:
Testing Export

Commit:
test export and general bugs

Commit:
update loan when update contact

matthewng1996 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113T-W17-2/main

# Conflicts:
#	src/main/java/wallet/logic/command/ListCommand.java

GeraldLimWeeKoon committed:
FIxed: Edit loan date bugs and complexity
Fixed: Command History bugs

GeraldLimWeeKoon committed:
Fixed: checkstyle

matthewng1996 committed:
Merge pull request #96 from A0171206R/master

FIxed previous bugs

matthewng1996 committed:
Fix budget commands bug
Added more strings to budget-expense notifications

matthewng1996 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113T-W17-2/main

matthewng1996 committed:
Added Loan Table to ListCommand ie list 01/01/2019 will show both expenses and loans with the date

Commit:
get latest loan class

Commit:
fix checkstyle 1

Commit:
test tab

Commit:
use message_list_contacts

Commit:
add back fallthrough

Commit:
switch index to id in deletecommand

Commit:
helpData

kyang96 committed:
Added Category Enum

kyang96 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113T-W17-2/main

kyang96 committed:
Added view statistics feature to view monthly expenses in each category

Commit:
oop for help

Commit:
added author tags - hopefully correct

Commit:
add spacing before package

matthewng1996 committed:
Added JUnit test for Budget, BudgetParser, View, ViewCommandParser
Added authors for RepoSense

Commit:
Merge pull request #97 from Xdecosee/master

Export (not completed) and bug fix for != ""

matthewng1996 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113T-W17-2/main

matthewng1996 committed:
Slight fix RepoSense

matthewng1996 committed:
Codacy Fix

GeraldLimWeeKoon committed:
added: undo,redo halfway

GeraldLimWeeKoon committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113T-W17-2/main

kyang96 committed:
Added AddCommandParserTest class

kyang96 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113T-W17-2/main

# Conflicts:
#	src/main/java/wallet/logic/command/EditCommand.java

kyang96 committed:
Removed DukeTest.java

kyang96 committed:
Merge pull request #98 from matthewng1996/master

Notifications (Budget with Expenses)

kyang96 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113T-W17-2/main

kyang96 committed:
Updated output of list recurring expenses to table form.

kyang96 committed:
Merge pull request #99 from kyang96/master

Added Category Enum and view monthly expenses based on categories

matthewng1996 committed:
Delete outdated Dev Guide

Will be replaced with latest dev guide

matthewng1996 committed:
Latest DG as of 24th October 2019

matthewng1996 committed:
Minor fix DG

matthewng1996 committed:
Latest DG as of 24th October

matthewng1996 committed:
Remove outdated UG

matthewng1996 committed:
Updated UG as of 24th October

matthewng1996 committed:
Convert adocs to markdown

matthewng1996 committed:
Fix #1

Minor fix

Commit:
Export Expenses and Global Error Msg

Commit:
fix checkstyle and -1 logic

Commit:
replace repeat() since only java 11 support

Commit:
amount and date swap

Commit:
add dollar sign to headers

GeraldLimWeeKoon committed:
Added: undo working properly

GeraldLimWeeKoon committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113T-W17-2/main

GeraldLimWeeKoon committed:
added: RedoCommand class
edited: Refactored all LogicManager.getWallet() --> LogicManager.getWalletList().getWalletList().get(LogicManager.getWalletList().getState()) as there is no longer just a single Wallet object. There are multiple Wallet objects which hold a particular state.

GeraldLimWeeKoon committed:
Added: //@@A0171206R to claim my code

matthewng1996 committed:
Merge pull request #101 from Xdecosee/master

Export Table

matthewng1996 committed:
Pie chart is now able to read existing Expenses and draw them.
Changed "view pie" to "view stats"
Commented off ui.printLine due to race condition existing from printline.

matthewng1996 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113T-W17-2/main

matthewng1996 committed:
Remove unused method.

Commit:
working on import

matthewng1996 committed:
Pie chart to take in Arraylist of expenses parameter to draw pie charts for selected months
Debug: Rearrange condition in View Command
Debug: Expenses table printed twice, removed a few lines to debug.
Checkstyle fix

Commit:
import done

Commit:
sample files

GeraldLimWeeKoon committed:
Added: basic junit testing

GeraldLimWeeKoon committed:
Merge pull request #102 from matthewng1996/master

Working PieChart

GeraldLimWeeKoon committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113T-W17-2/main

Commit:
add javadoc for import and export

Commit:
add javadoc for import and export

Commit:
minor fixes

Commit:
junit tests

Commit:
junit tests

Commit:
fix checkstyle

Commit:
fix checkstyle

Commit:
checkstyle 2

Commit:
checkstyle 2

Commit:
checkstyle 3

Commit:
checkstyle 3

Commit:
checkstyle 4

Commit:
checkstyle 4

GeraldLimWeeKoon committed:
Merge pull request #103 from A0171206R/master

Added a undo/redo and Junit for loans

GeraldLimWeeKoon committed:
Merge branch 'master' into master

GeraldLimWeeKoon committed:
Merge pull request #104 from Xdecosee/master

Import Table

Commit:
resolve conflicts and getwallet()

Commit:
fix checkstyle

Commit:
checkstyle 1

Commit:
Merge pull request #105 from Xdecosee/master

Change logicmanager.getWallet() method

matthewng1996 committed:
Currency Conversion #1

matthewng1996 committed:
checkstyle fix for Travis

matthewng1996 committed:
Add Junit Testing for CurrencyCommand and some checkstyle fix

matthewng1996 committed:
Travis has some error since test all passed on my side.

Commit:
Create project portfolio

Commit:
Delete project portfolio

Commit:
Create XDECOSEE-PORTFOLIO.md

Commit:
Update XDECOSEE-PORTFOLIO.md

Commit:
Update XDECOSEE-PORTFOLIO.md

Commit:
Update XDECOSEE-PORTFOLIO.md

Commit:
Update XDECOSEE-PORTFOLIO.md

Commit:
Update XDECOSEE-PORTFOLIO.md

Commit:
Update XDECOSEE-PORTFOLIO.md

Commit:
Update XDECOSEE-PORTFOLIO.md

Commit:
Update XDECOSEE-PORTFOLIO.md

Commit:
Update XDECOSEE-PORTFOLIO.md

Commit:
Update XDECOSEE-PORTFOLIO.md

Commit:
Update XDECOSEE-PORTFOLIO.md

Commit:
Update XDECOSEE-PORTFOLIO.md

kyang96 committed:
Fix error in edit command when identifiers are in wrong order.

kyang96 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113T-W17-2/main

# Conflicts:
#	src/main/java/wallet/logic/command/ListCommand.java
#	src/main/java/wallet/ui/Ui.java

Commit:
remove md

Commit:
update help docs and fix export bug

Commit:
update indentation

Commit:
fix travis

Commit:
Merge pull request #107 from Xdecosee/master

Update help docs to v1.3

matthewng1996 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113T-W17-2/main

kyang96 committed:
Added bar graph when viewing statistics

matthewng1996 committed:
Merge pull request #109 from kyang96/master

Added bar graph when viewing statistics

GeraldLimWeeKoon committed:
Merge pull request #106 from matthewng1996/master

Currency Conversion #1

matthewng1996 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113T-W17-2/main

matthewng1996 committed:
Fixed Pie Chart bug

matthewng1996 committed:
Added Symbol to table before Pie Chart

matthewng1996 committed:
TRAVIS

matthewng1996 committed:
travis #2

matthewng1996 committed:
Travis + Codacy

GeraldLimWeeKoon committed:
Merge pull request #113 from matthewng1996/master

Fixed Pie Chart bug

GeraldLimWeeKoon committed:
Added: New package exception to handle all custom exceptions

matthewng1996 committed:
DG/UG Updated as of 1/11/2019

matthewng1996 committed:
Merge pull request #115 from A0171206R/master

Added Custom Exceptions

matthewng1996 committed:
Added custom exceptions for budget and view command

matthewng1996 committed:
Added sorting functionalities for list proper
Rearrange Category to alphabetical order for sorting purposes
Added methods to sort in both LoanList and ExpenseList

matthewng1996 committed:
Updated UG as of 2nd November 2019

This update consists of the following:
1) Document fix from the bugs found in PE dry run
2) Added missing screenshots in List/Expenses/Loans
3) Re-format some parts of documents 
         a) Each feature should have this format in the document:
                   Format....
                   Example....
                   [Screenshot]
                   Screenshot label.

matthewng1996 committed:
Codacy fix

Commit:
Merge pull request #127 from matthewng1996/master

Added sorting functions in List

GeraldLimWeeKoon committed:
small changes to Reminder I/O

GeraldLimWeeKoon committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113T-W17-2/main

GeraldLimWeeKoon committed:
Added: Minimum time of 10 minutues when setting reminders
Fixed ListCommand: 'list loan' insteaed of 'list loans'
Changed: the error of the InsufficientParameter in AddCommandParser

kyang96 committed:
Added more JUnit testing for ExpenseList class

kyang96 committed:
Added more useful error messages when adding/deleting expenses.

kyang96 committed:
Fix bug where invalid category is accepted.

GeraldLimWeeKoon committed:
Pleasing travis

matthewng1996 committed:
Merge pull request #131 from A0171206R/master

Addressing bug issues

Commit:
handle exception for budget year/month

Commit:
change to loan and expense

Commit:
sort contacts by name

Commit:
export file with datetime

Commit:
travis fix

Commit:
change back to camel case

kyang96 committed:
Merge pull request #132 from Xdecosee/master

Enhancement to contact and export

kyang96 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113T-W17-2/main

# Conflicts:
#	src/main/java/wallet/logic/parser/AddCommandParser.java

kyang96 committed:
Fixed checkstyle issues

Commit:
minor changes

Commit:
Merge pull request #133 from kyang96/master

Added more JUnit testing and error handling

Commit:
redo help oop

Commit:
help author:

Commit:
redo oop for import

matthewng1996 committed:
ListCommandTest replace CurrencyTest

matthewng1996 committed:
Checkstyle fix

GeraldLimWeeKoon committed:
Changed: Changed WalletCMI back to WalletCLI

GeraldLimWeeKoon committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113T-W17-2/main

Commit:
travis fix

Commit:
codacy fix

Commit:
codacy fix 2

Commit:
codact fix 3

Commit:
coday fix 4

Commit:
travis fix 2

Commit:
remove exception import

Commit:
remove import

Commit:
codacy final fix

Commit:
deletecomand junit for contact

Commit:
format junit

Commit:
update command syntax

Commit:
remove import

Commit:
Merge pull request #135 from matthewng1996/master

Added JUnit Testing for ListCommand

kyang96 committed:
Merge pull request #136 from Xdecosee/master

Some OOP changes

kyang96 committed:
Modified main logo and added more exception handling for edit and view commands.

kyang96 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113T-W17-2/main

kyang96 committed:
Changed rate of recurrence from String to Enum.

Commit:
Merge pull request #137 from kyang96/master

Added RecurrenceRate Enum and more exception handling.

Commit:
change example files

Commit:
Merge pull request #138 from Xdecosee/master

Change example export files

matthewng1996 committed:
Budget amount that takes into account of expenses is now separated from set budget.
Set budget is therefore a fixed value now.
No change in commands (Does not affect any existing commands). Only new features are added.

matthewng1996 committed:
Fixed Junit Test for budget

matthewng1996 committed:
Merge pull request #139 from matthewng1996/master

Fixed amount for budget

GeraldLimWeeKoon committed:
removed: storage error message temporarily for bug fix

GeraldLimWeeKoon committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113T-W17-2/main

GeraldLimWeeKoon committed:
added: loan model testing

kyang96 committed:
Added Junit test for ViewCommand and ChartThread

kyang96 committed:
Added Junit test for DeleteCommandParser

GeraldLimWeeKoon committed:
Merge pull request #141 from kyang96/master

More JUnit test

GeraldLimWeeKoon committed:
added: generate command to put in default data

GeraldLimWeeKoon committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113T-W17-2/main

GeraldLimWeeKoon committed:
Fixed: checkstyle issues regarding GenerateCommand class

GeraldLimWeeKoon committed:
fixed: codacy problems

GeraldLimWeeKoon committed:
fixed: javadocs issue

matthewng1996 committed:
Merge pull request #140 from A0171206R/master

Minor additions and temporary solution for storage bug.

matthewng1996 committed:
Fixed PieChart Bug

GeraldLimWeeKoon committed:
Merge pull request #142 from matthewng1996/master

Fixed PieChart Bug

Commit:
bug fixes and logic errors

Commit:
fixing the travis

Commit:
change error msg

kyang96 committed:
Merge pull request #143 from Xdecosee/master

bug fixes and logic errors

kyang96 committed:
Added more testing for recurring expenses.

Commit:
Standardize Exception and other matters

Commit:
test error

Commit:
travis fix

Commit:
travis fix 2

Commit:
remove comment

kyang96 committed:
Merge pull request #144 from Xdecosee/master

Standardize Exception and other matters

kyang96 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113T-W17-2/main

matthewng1996 committed:
Fixed AddCommand not to save budget
Budget to take into account expenses that are accounted/not accounted for based on user's input

kyang96 committed:
Fixed bug where storage error message keep appearing when there is no data.

kyang96 committed:
Merge pull request #145 from matthewng1996/master

Fixed AddCommand not to save budget

kyang96 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113T-W17-2/main

kyang96 committed:
Fix README.md

kyang96 committed:
Updated Ui.png

kyang96 committed:
Refactor code.

Commit:
fix budget

Commit:
remove star

Commit:
Merge pull request #146 from kyang96/master

Refactor code and fixed storage error messages.

Commit:
codacy

Commit:
travis

kyang96 committed:
Merge pull request #147 from Xdecosee/master

Copy Budget codes into Import and Export

Commit:
remove unused 1

Commit:
remove unused 2

Commit:
Merge pull request #148 from Xdecosee/master

Remove unused

kyang96 committed:
Fixed bug in EditCommand

Commit:
handle space and tab

GeraldLimWeeKoon committed:
fixed: loan bugs eg. "/lbbb" no longer valid

Commit:
handle space and tab at parsermanager

kyang96 committed:
Merge pull request #149 from Xdecosee/master

handle space and tab

kyang96 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113T-W17-2/main

# Conflicts:
#	src/main/java/wallet/logic/parser/AddCommandParser.java

Commit:
Merge pull request #150 from kyang96/master

Fixed bug in EditCommand

GeraldLimWeeKoon committed:
added: better exception handling when adding loans.

GeraldLimWeeKoon committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113T-W17-2/main

kyang96 committed:
Merge pull request #151 from A0171206R/master

Super duper extra secure sanitisation when adding loans

matthewng1996 committed:
Fixed bug to budget when changing currency

GeraldLimWeeKoon committed:
fixed bug for: "done loan 1 2 3 4"

kyang96 committed:
Fix EditCommand error message

kyang96 committed:
Merge pull request #152 from matthewng1996/master

Fixed bug to budget when changing currency

kyang96 committed:
Merge pull request #153 from A0171206R/master

Minor bug fix

GeraldLimWeeKoon committed:
Merge pull request #154 from kyang96/master

Fix EditCommand error message

Commit:
fix edit contact bug

Commit:
codacy fix

Commit:
Merge pull request #155 from Xdecosee/master

Fix edit contact bug

Commit:
change files

Commit:
Merge pull request #156 from Xdecosee/master

change files

matthewng1996 committed:
Latest/Final DG and UG

matthewng1996 committed:
Remove outdated UG

matthewng1996 committed:
Remove outdated DG

GeraldLimWeeKoon committed:
fixed: minor bugs in printing loans and reminders when contact does not have a phone number

GeraldLimWeeKoon committed:
pleasing travis

GeraldLimWeeKoon committed:
Merge pull request #157 from A0171206R/master

Fixed: Listing loan I/O bug
